tenable/io/v3/vm/plugins.py

```diff
@@ -0,0 +1,156 @@
+        to the plugins.  Effectively what we doing is generating a dictionary of
+        2 subdictionaries.  Each one of these is a simple hash table allowing
+        the iterator to resolve the name of the family by ID and the family ID
+        by the plugin membership.  This information is currently lacking in the
+        plugin listing output and was requested by a customer.
+
+        .. note::
+            This currently seems to add about 7-10 seconds before the first item
+            is returned, as it seems to take this long to generate the data. We
+            can focus on reducing this time later on with the introduction of
+            multi-threaded iterators && async API calls.
+        """
+        self._maptable = {
+            'plugins': dict(),
+            'families': dict()
+        }
+        for family in self._api.plugins.families():
+            self._maptable['families'][family['id']] = family['name']
+
+        for fam_id in self._maptable['families'].keys():
+            for plugin in self._api.plugins.family_details(fam_id)['plugins']:
+                self._maptable['plugins'][plugin['id']] = fam_id
+
+    def next(self):
+        item = super(PluginIterator, self).next()
+
+        # If the populate_maptable flag is set, then we will build the mappings.
+        if not self._maptable and self.populate_maptable:
+            self._populate_family_cache()
+
+        # If the maptable exists, then graft on the plugin family information
+        # on to to the item.
+        if self._maptable:
+            try:
+                fid = self._maptable['plugins'][item['id']]
+                item['family_id'] = fid
+                item['family_name'] = self._maptable['families'][fid]
+            except KeyError:
+                self._log.warning("plugin id {} not found in plugin family".format(item['id']))
+                item['family_id'] = None
+                item['family_name'] = None
+        return item
+
+
+class PluginsAPI(ExploreBaseEndpoint):
+    """
+    This will contain all methods related to plugins
+    """
+    _path = "plugins"
+    _conv_json = True
+
+    def families(self) -> List:
+        """
+        List the available plugin families.
+
+        :devportal:`plugins: families <plugins-families>`
+
+        Returns:
+            List:
+                List of plugin family resource records.
+
+        Examples:
+            >>> for family in tio.plugins.families():
+            ...     pprint(family)
+        """
+        return NotImplementedError("Search and Filter functionality will be updated later.")
+
+    def family_details(self, family_id: int) -> Dict:
+        """
+        Retrieve the details for a specific plugin family.
+
+        :devportal:`plugins: family-details plugins-family-details>`
+
+        Args:
+            family_id:
+                The plugin family unique identifier.
+
+        Returns:
+            Dict:
```

Comment:
Update the return type everywhere applicable based on changes done here:
https://github.com/dhvanilshah-tenable/pyTenable/pull/18/files#diff-3e4c648573b0079ec2ce6205364bd8cffb02a5d4f321ee678e895620b30da60dL157